### DISCUSSION TITLE
"optional" subdir()?

### DISCUSSION THREAD

**Discussion by nils-werner** [mesonbuild/meson#13159] [Q&A] [answered]
Is there a way to implement an optional/not-required `subdir()`?

I would like to provide my team with a `.gitignore`d directory, like `workbench/`, where they can put short-lived code-experiments that interact with our codebase, but shouldn't permanently or accidentally end up in Git.

**Comment by nils-werner** [comment, marked as answer]
The more I thought about this the less I liked the "magic but unexpected behaviour", so I've decided making this a conscious choice by putting a parameter `enable_workbench` in `meson_options.txt`.

**Comment by eli-schwartz** [comment]
You could use:
```meson
fs_mod = import('fs')

if fs.exists('optional/meson.build')
    subdir('optional')
endif
```

Note that adding that subdir will not trigger a meson reconfigure so if users do that then they'll need to also manually reconfigure. Removing it, however, *will* trigger reconfiguration.
